lib/ostruct.rb

```diff
@@ -239,7 +239,10 @@ def freeze
         raise! ArgumentError, "wrong number of arguments (given #{len}, expected 1)", caller(1)
       end
       set_ostruct_member_value!(mname, args[0])
-    elsif len == 0
+    elsif len == 0 # and /\A[a-z_]\w*\z/ =~ mid #
+      if @table.key?(mid)
```

Comment:
I suppose we can remove this as well since `@table[mid]` will return nil already if the key isn't in the hash.